### PR TITLE
add bin helper functions

### PR DESCRIPTION
**Pull request by asizemore**
add helpers mentioned in https://github.com/VEuPathDB/microbiomeComputations/pull/43

after merge i can replace the plot.data versions of these functions

### CODE REVIEW

File: R/utils.R [hunk 1]
```diff
@@ -94,4 +94,74 @@ matchArg <- function(arg, choices) {
...
+}
+
+
+#' Which values are In the Bin?
+#' 
+#' whichInBin finds array values that fall within a given bin
+#' @param values numeric vector
+#' @param binString a string that defines a numeric bin. For example '[1, 2)'
+#' @export
+whichInBin <- function(values, binString) {
```
Comment (d-callan) [inline R/utils.R:106]: i feel like we could name this a bit better..

Comment (d-callan) [reply R/utils.R:106]: also why are we using binString and not the Bin class?

File: R/utils.R [hunk 2]
```diff
@@ -94,4 +94,74 @@ matchArg <- function(arg, choices) {
...
+}
+
+
+#' Find Bin Start
+#' 
+#' findBinStart finds the start value of a given bin.
+#' @param x string that represents a bin. Ex: '[1, 2)'
+#' @param addTimeZone boolean determines if time zone should be added to dates. Default is FALSE.
+#' @export
+findBinStart <- function(x, addTimeZone = c(FALSE, TRUE)) {
```
Comment (d-callan) [inline R/utils.R:121]: feel like this should be called findBinStartFromString or something

Comment (d-callan) [reply R/utils.R:121]: im also curious why we need this? can the frontend not pass the exact bin start and ends? why are we getting a string?

**Comment by asizemore** [comment]
Officially using real grown up Bins and BinLists!
@d-callan ready for another look!

I did have trouble with naming. Maybe isValueInBin? That answer would at least be T/F. Starting with "which" isn't so bad but it doesn't work the way `which()` does because it doesn't return indices. But i often want it to return a logical vector instead of the indices so...

**Review by d-callan — APPROVED** [review]
im ok w these changes and am happy to approve. i suspect we should be adding a `Comparator` class to this repo as well, and should consider renaming `Bin` since it doesnt really need to have anything but a label. but those can be separate PRs/ issues.

**Pull request merged**